satpy/tests/reader_tests/test_eum_l2_grib.py

```diff
@@ -97,38 +132,7 @@ def test_seviri_data_reading(self, da_, xr_):
 
                 dataset_id = make_dataid(name="dummmy", resolution=3000)
 
-                # Checks that the codes_grib_multi_support_on function has been called
-                self.ec_.codes_grib_multi_support_on.assert_called()
-
-                # Restarts the id generator and clears the call history
-                fake_gid_generator = (i for i in FAKE_GID)
-                self.ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
-                self.ec_.codes_grib_new_from_file.reset_mock()
-                self.ec_.codes_release.reset_mock()
-
-                # Checks the correct execution of the get_dataset function with a valid parameter_number
-                valid_dataset = self.reader.get_dataset(dataset_id, {"parameter_number": 30})
-                # Checks the correct file open call
-                mock_file.assert_called_with("test.grib", "rb")
-                # Checks that the dataset has been created as a DataArray object
-                assert valid_dataset._extract_mock_name() == "xr.DataArray()"
-                # Checks that codes_release has been called after each codes_grib_new_from_file call
-                # (except after the last one which has returned a None)
-                assert self.ec_.codes_grib_new_from_file.call_count == self.ec_.codes_release.call_count + 1
-
-                # Restarts the id generator and clears the call history
-                fake_gid_generator = (i for i in FAKE_GID)
-                self.ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
-                self.ec_.codes_grib_new_from_file.reset_mock()
-                self.ec_.codes_release.reset_mock()
-
-                # Checks the correct execution of the get_dataset function with an invalid parameter_number
-                invalid_dataset = self.reader.get_dataset(dataset_id, {"parameter_number": 50})
-                # Checks that the function returns None
-                assert invalid_dataset is None
-                # Checks that codes_release has been called after each codes_grib_new_from_file call
-                # (except after the last one which has returned a None)
-                assert self.ec_.codes_grib_new_from_file.call_count == self.ec_.codes_release.call_count + 1
+                self.common_checks(mock_file, dataset_id)
 
                 # Checks the basic data reading
                 assert REPEAT_CYCLE_DURATION == 15
```

Comment:
Not sure we really need this assert? It's just a constant imported from `eum_base.py`, and now we also have `REPEAT_CYCLE_DURATION_RSS`. Perhaps rather verify the `start_time` and `end_time`, which would be more useful.

In this case we have `ssp_lon=9.5`, so `start_time` should be 2019-10-20 17:45 and `end_time` 2019-10-20 17:50` as far as I can tell from the fake message.

Reply:
Solved